pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/nonpepperkit/NonPepperStatusKitService.java

```diff
@@ -0,0 +1,149 @@
+            return new KitResponseError(KitResponse.UNKNOWN_STUDY);
+        }
+        BookmarkDto kitPagingThreshold = new BookmarkDao().getBookmarkByInstance(STATUS_KIT_THRESHOLD)
+                .orElse(new BookmarkDto.Builder(20, STATUS_KIT_THRESHOLD).build()); // use a threshold of 20 if none provided
+
+        // get all the kits
+        ResultSet foundKitResults = kitDao.getKitsInDatabaseByInstanceId(ddpInstance);
+        return processNonPepperKitStatusFromResultSet(foundKitResults);
+    }
+
+    public KitResponse getKitsBasedOnJuniperKitId(String juniperKitId) {
+        if (StringUtils.isBlank(juniperKitId)) {
+            return new KitResponseError(KitResponse.MISSING_JUNIPER_KIT_ID);
+        }
+        // get the kit with that juniperKitId
+        ResultSet foundKitResults = kitDao.getKitsByJuniperKitId(juniperKitId);
+        return processNonPepperKitStatusFromResultSet(foundKitResults);
+
+    }
+
+    public KitResponse getKitsBasedOnParticipantId(String participantId) {
+        if (StringUtils.isBlank(participantId)) {
+            return new KitResponseError(KitResponse.MISSING_JUNIPER_PARTICIPANT_ID);
+        }
+        // get the kit with that participantId
+        ResultSet foundKitResults = kitDao.getKitsByParticipantId(participantId);
+        return processNonPepperKitStatusFromResultSet(foundKitResults);
+    }
+
+    public KitResponse getKitsFromKitIds(String[] kitIdsArray) {
+        // get the kits with the given kit ids
+        try {
+            ArrayList<NonPepperKitStatus> list =  kitDao.getKitsByKitIdArray(kitIdsArray, this);
+            StatusKitResponse statusKitResponse = new StatusKitResponse(list);
+            return statusKitResponse;
+
+        } catch (Exception e) {
+            log.error("Error getting kits by an array of kit ids", e);
+            return new KitResponseError(KitResponse.DSM_ERROR);
+        }
+    }
+
+    private KitResponse processNonPepperKitStatusFromResultSet(ResultSet foundKitResults) {
+        ArrayList<NonPepperKitStatus> list = selectAllNonPepperKitStatus(foundKitResults);
+        StatusKitResponse statusKitResponse = new StatusKitResponse(list);
+        return statusKitResponse;
+    }
+
+    public ArrayList<NonPepperKitStatus> selectAllNonPepperKitStatus(ResultSet foundKitResults) {
+        HashMap<Integer, UserDto> users = UserDao.selectAllUsers();
+        ArrayList<NonPepperKitStatus> kits = new ArrayList<>();
+        try {
+            while (foundKitResults.next()) {
+                kits.add(createStatusForKit(foundKitResults, users));
+            }
+
+        } catch (SQLException e) {
+            log.error("Some Error getting the status of kits", e);
+        }
+        return kits;
+    }
+
+    private NonPepperKitStatus createStatusForKit(ResultSet foundKitResults, HashMap<Integer, UserDto> users) throws SQLException {
+        String labelUser = getUserEmailForFields(foundKitResults.getString(DBConstants.LABEL_BY), users);
```

Comment:
As mentioned in my comments for the DAO, it feels like this code belongs in the DAO.